src/barchart_split/BarChartSplit.js

```diff
@@ -0,0 +1,228 @@
+      .attr('width', this.scaleColumn.bandwidth())
+      .attr('height', this.scaleY.bandwidth())
+      .attr('opacity', '.2')
+      .attr('fill', 'var(--gv-grey)');
+
+    gColumn.selectAll('.bar-chart-small-overlying')
+      .data((d) => d.values)
+      .join('rect')
+      .attr('class', 'bar-chart-small-overlying')
+      .attr('x', 0)
+      .attr('y', d => this.scaleY(d[this.yAxisProp]))
+      .attr('width', d => this.arrayOfScales[this.arrayOfXaxisProp.findIndex(element => element === d[this.xAxisProp])](d[this.countProp]))
+      .attr('height', this.scaleY.bandwidth())
+      .attr("fill", d => this.scaleColor(d[this.xAxisProp]))
+
+    gColumn
+    .selectAll(".label")
+    .data(d => d.values)
+    .join('text')
+    .attr('class', 'label')
+    .text(d => d[this.countProp])
+    .each((d, i, element) => {
+      const xValue = this.arrayOfScales[this.arrayOfXaxisProp.findIndex(element => element === d[this.xAxisProp])](d[this.countProp])
+      const xMax = select(element[i])._groups[0][0].getBBox().width;
+     if (xValue < xMax) {
+        select(element[i])
+          .attr('x', xValue + 10)
+          .attr("fill","#000")
+      } else {
+        select(element[i])
+          .attr("fill","#fff")
+          .attr('x', 0)
+      }
+      select(element[i])
+        .attr('y', this.scaleY(d[this.yAxisProp]) + this.scaleY.bandwidth() / 2)
+        .attr('dy', '0.33em')
+        .attr('dx', 4);
+    });
+
+  }
+
+  yAxis(g) {
+    g.call(axisLeft(this.scaleY));
+
+    // remove baseline
+    g.select(".domain").remove();
+
+    // remove default formats
+    g.attr("font-family", null).attr("font-size", null);
+
+    // change line style defaults
+  }
+
+  async setData(data) {
+    this.rawData = data
+    this.data = this.parse(data)
+    this.arrayOfXaxisProp = [...new Set(this.data.map(d => d[this.xAxisProp]))];
+
+    // only set the color scale, as of the first time you get the data
+    if (!this.scaleColor) {
+      this.setColorScale();
+    }
+
+    this.build();
+  }
+
+  setScales() {
+
+    this.svg
+      .attr("width", `${this.width + this.margin.left + this.margin.right}`)
+      .attr("height", `${this.height + this.margin.top + this.margin.bottom}`);
+
+    this.scaleY = scaleBand()
+      .domain(this.data.map(d => d[this.yAxisProp]).reverse())
+      .range([this.height, 0]).padding(0.4);
+
+    this.scaleColumn = scaleBand()
+      .domain([...new Set(this.data.map(d => d[this.xAxisProp]))])
+      .range([0, this.width]).paddingInner(0.1);
+
+  }
+
+  //Create an array of scales, one scale for each of the columns.
+  createArrayOfScales() {
+    this.arrayOfScales = []
+    for (let scale of this.arrayOfXaxisProp) {
+      let elementScale = scaleLinear()
+        .range([0, this.scaleColumn.bandwidth()])
+        .domain([0, max(this.data.filter(element => scale.includes(element[this.xAxisProp])), (d) => d[this.countProp])]).nice();
+        this.arrayOfScales.push(elementScale)
+    }
```

Comment:
Teel me if this don't do the same:

```suggestion
    this.arrayOfScales = scaleLinear()
        .range([0, this.scaleColumn.bandwidth()])
        .domain([0, max(this.data.filter(element => scale.includes(element[this.xAxisProp])), (d) => d[this.countProp])]).nice();
```

Tell me again, if you can't put this code block in the setScales function, saving a declaration